src/main/java/ru/kh/redis/repositories/CacheRepository.java

```diff
@@ -0,0 +1,132 @@
+package ru.kh.redis.repositories;
+
+import org.springframework.stereotype.Repository;
+import ru.kh.redis.models.Key;
+import ru.kh.redis.models.entities.StoredEntity;
+import ru.kh.redis.dto.keysDto.KeyExpireDto;
+
+import java.lang.ref.SoftReference;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+@Repository
+public class CacheRepository {
+
+    private SoftReference<ConcurrentHashMap<Key, StoredEntity>> cacheMapRef =
+            new SoftReference<>(new ConcurrentHashMap<>());
+
+    public StoredEntity getValue(Key key) {
+        return (StoredEntity) function(inputKey -> {return cacheMapRef.get().get(inputKey);}, key);
```

Comment:
почему не expression lambda?
```suggestion
        return (StoredEntity) function(inputKey ->cacheMapRef.get().get(inputKey), key);
```

Reply:
в остальных местах аналогично